helixbass/tsc-rust
# Type-check union of string literals

In this PR:
- type-check variable declaration of union-of-string-literals type

To test:
If you run `cargo run tmp.tsx` against a source file containing eg `const a: 'foo' | 'bar' = 'baz'` you should see a diagnostic debug-logged with message `Type '"baz"' is not assignable to type '"foo" | "bar"'.`
If you run `cargo run tmp.tsx` against a source file containing eg `const a: 'foo' | 'bar' = 'bar'` you shouldn't see any diagnostics debug-logged

Based on `parse-union-of-string-literals`
